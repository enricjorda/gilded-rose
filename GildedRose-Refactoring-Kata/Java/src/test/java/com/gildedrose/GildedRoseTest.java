package com.gildedrose;

import org.approvaltests.Approvals;
import org.approvaltests.combinations.CombinationApprovals;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;

public class GildedRoseTest {

    @Test
    public void foo() {
        CombinationApprovals.verifyAllCombinations(
                this::updateQualityToString,
                new String[]{
                        "Regular",
                        "Aged Brie",
                        "Backstage passes to a TAFKAL80ETC concert",
                        "Sulfuras, Hand of Ragnaros"},
                new Integer[]{10,11,5,6,-5,0},
                new Integer[]{0,50,49});
    }

    private String updateQualityToString(String name, int sellIn, int quality) {
        Item[] items = new Item[] { new Item(name, sellIn, quality) };
        GildedRose app = new GildedRose(items);
        app.updateQuality();
        return app.items[0].toString();
    }

}
