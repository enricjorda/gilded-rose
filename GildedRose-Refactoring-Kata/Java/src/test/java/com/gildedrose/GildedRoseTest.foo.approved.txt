[Regular, 10, 0] => Regular, 9, 0 
[Regular, 10, 50] => Regular, 9, 49 
[Regular, 10, 49] => Regular, 9, 48 
[Regular, 11, 0] => Regular, 10, 0 
[Regular, 11, 50] => Regular, 10, 49 
[Regular, 11, 49] => Regular, 10, 48 
[Regular, 5, 0] => Regular, 4, 0 
[Regular, 5, 50] => Regular, 4, 49 
[Regular, 5, 49] => Regular, 4, 48 
[Regular, 6, 0] => Regular, 5, 0 
[Regular, 6, 50] => Regular, 5, 49 
[Regular, 6, 49] => Regular, 5, 48 
[Regular, -5, 0] => Regular, -6, 0 
[Regular, -5, 50] => Regular, -6, 48 
[Regular, -5, 49] => Regular, -6, 47 
[Regular, 0, 0] => Regular, -1, 0 
[Regular, 0, 50] => Regular, -1, 48 
[Regular, 0, 49] => Regular, -1, 47 
[Aged Brie, 10, 0] => Aged Brie, 9, 1 
[Aged Brie, 10, 50] => Aged Brie, 9, 50 
[Aged Brie, 10, 49] => Aged Brie, 9, 50 
[Aged Brie, 11, 0] => Aged Brie, 10, 1 
[Aged Brie, 11, 50] => Aged Brie, 10, 50 
[Aged Brie, 11, 49] => Aged Brie, 10, 50 
[Aged Brie, 5, 0] => Aged Brie, 4, 1 
[Aged Brie, 5, 50] => Aged Brie, 4, 50 
[Aged Brie, 5, 49] => Aged Brie, 4, 50 
[Aged Brie, 6, 0] => Aged Brie, 5, 1 
[Aged Brie, 6, 50] => Aged Brie, 5, 50 
[Aged Brie, 6, 49] => Aged Brie, 5, 50 
[Aged Brie, -5, 0] => Aged Brie, -6, 2 
[Aged Brie, -5, 50] => Aged Brie, -6, 50 
[Aged Brie, -5, 49] => Aged Brie, -6, 50 
[Aged Brie, 0, 0] => Aged Brie, -1, 2 
[Aged Brie, 0, 50] => Aged Brie, -1, 50 
[Aged Brie, 0, 49] => Aged Brie, -1, 50 
[Backstage passes to a TAFKAL80ETC concert, 10, 0] => Backstage passes to a TAFKAL80ETC concert, 9, 2 
[Backstage passes to a TAFKAL80ETC concert, 10, 50] => Backstage passes to a TAFKAL80ETC concert, 9, 50 
[Backstage passes to a TAFKAL80ETC concert, 10, 49] => Backstage passes to a TAFKAL80ETC concert, 9, 50 
[Backstage passes to a TAFKAL80ETC concert, 11, 0] => Backstage passes to a TAFKAL80ETC concert, 10, 1 
[Backstage passes to a TAFKAL80ETC concert, 11, 50] => Backstage passes to a TAFKAL80ETC concert, 10, 50 
[Backstage passes to a TAFKAL80ETC concert, 11, 49] => Backstage passes to a TAFKAL80ETC concert, 10, 50 
[Backstage passes to a TAFKAL80ETC concert, 5, 0] => Backstage passes to a TAFKAL80ETC concert, 4, 3 
[Backstage passes to a TAFKAL80ETC concert, 5, 50] => Backstage passes to a TAFKAL80ETC concert, 4, 50 
[Backstage passes to a TAFKAL80ETC concert, 5, 49] => Backstage passes to a TAFKAL80ETC concert, 4, 50 
[Backstage passes to a TAFKAL80ETC concert, 6, 0] => Backstage passes to a TAFKAL80ETC concert, 5, 2 
[Backstage passes to a TAFKAL80ETC concert, 6, 50] => Backstage passes to a TAFKAL80ETC concert, 5, 50 
[Backstage passes to a TAFKAL80ETC concert, 6, 49] => Backstage passes to a TAFKAL80ETC concert, 5, 50 
[Backstage passes to a TAFKAL80ETC concert, -5, 0] => Backstage passes to a TAFKAL80ETC concert, -6, 0 
[Backstage passes to a TAFKAL80ETC concert, -5, 50] => Backstage passes to a TAFKAL80ETC concert, -6, 0 
[Backstage passes to a TAFKAL80ETC concert, -5, 49] => Backstage passes to a TAFKAL80ETC concert, -6, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 0] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 50] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 49] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Sulfuras, Hand of Ragnaros, 10, 0] => Sulfuras, Hand of Ragnaros, 10, 0 
[Sulfuras, Hand of Ragnaros, 10, 50] => Sulfuras, Hand of Ragnaros, 10, 50 
[Sulfuras, Hand of Ragnaros, 10, 49] => Sulfuras, Hand of Ragnaros, 10, 49 
[Sulfuras, Hand of Ragnaros, 11, 0] => Sulfuras, Hand of Ragnaros, 11, 0 
[Sulfuras, Hand of Ragnaros, 11, 50] => Sulfuras, Hand of Ragnaros, 11, 50 
[Sulfuras, Hand of Ragnaros, 11, 49] => Sulfuras, Hand of Ragnaros, 11, 49 
[Sulfuras, Hand of Ragnaros, 5, 0] => Sulfuras, Hand of Ragnaros, 5, 0 
[Sulfuras, Hand of Ragnaros, 5, 50] => Sulfuras, Hand of Ragnaros, 5, 50 
[Sulfuras, Hand of Ragnaros, 5, 49] => Sulfuras, Hand of Ragnaros, 5, 49 
[Sulfuras, Hand of Ragnaros, 6, 0] => Sulfuras, Hand of Ragnaros, 6, 0 
[Sulfuras, Hand of Ragnaros, 6, 50] => Sulfuras, Hand of Ragnaros, 6, 50 
[Sulfuras, Hand of Ragnaros, 6, 49] => Sulfuras, Hand of Ragnaros, 6, 49 
[Sulfuras, Hand of Ragnaros, -5, 0] => Sulfuras, Hand of Ragnaros, -5, 0 
[Sulfuras, Hand of Ragnaros, -5, 50] => Sulfuras, Hand of Ragnaros, -5, 50 
[Sulfuras, Hand of Ragnaros, -5, 49] => Sulfuras, Hand of Ragnaros, -5, 49 
[Sulfuras, Hand of Ragnaros, 0, 0] => Sulfuras, Hand of Ragnaros, 0, 0 
[Sulfuras, Hand of Ragnaros, 0, 50] => Sulfuras, Hand of Ragnaros, 0, 50 
[Sulfuras, Hand of Ragnaros, 0, 49] => Sulfuras, Hand of Ragnaros, 0, 49 
